persistent/picklecache.py

```diff
@@ -229,36 +307,104 @@ def debug_info(self):
     def update_object_size_estimation(self, oid, new_size):
         """ See IPickleCache.
         """
-        pass #pragma NO COVER
+        value = self.data.get(oid)
+        if value is not None:
+            # Recall that while the argument is given in bytes,
+            # we have to work with 64-block chunks (plus one)
+            # to match the C implementation. Hence the convoluted
+            # arithmetic
+            new_size_in_24 = _estimated_size_in_24_bits(new_size)
+            p_est_size_in_24 = value._Persistent__size
+            new_est_size_in_bytes = (new_size_in_24 - p_est_size_in_24) * 64
+
+            self.total_estimated_size += new_est_size_in_bytes
 
-    cache_size = property(lambda self: self.target_size)
+    cache_size = property(lambda self: self.target_size,
+                          lambda self, nv: setattr(self, 'target_size', nv))
```

Comment:
Because ZODB writes to it. E.g., https://github.com/NextThought/ZODB/blob/pypy/src/ZODB/DB.py#L854 :
```python
    def setHistoricalCacheSize(self, size):
        self._a()
        try:
            self._historical_cache_size = size
            def setsize(c):
                c._cache.cache_size = size
            self.historical_pool.map(setsize)
        finally:
            self._r()
```